core/src/main/java/org/apache/calcite/adapter/enumerable/EnumerableBatchNestedLoopJoin.java

```diff
@@ -55,6 +55,7 @@
 public class EnumerableBatchNestedLoopJoin extends Join implements EnumerableRel {
 
   private final ImmutableBitSet requiredColumns;
+  private final Join originalJoin;
```

Comment:
Keeping a RelNode in another RelNode is a strange pattern that we shall better avoid. Other than that adding attributes may necessitate touching the digest which may further complicate the situation.